Userland/Applications/FileManager/main.cpp

```diff
@@ -503,17 +503,10 @@ ErrorOr<int> run_in_desktop_mode()
         }
     };
 
-    struct BackgroundWallpaperListener : Config::Listener {
-        virtual void config_string_did_change(String const& domain, String const& group, String const& key, String const& value) override
-        {
-            if (domain == "WindowManager" && group == "Background" && key == "Wallpaper")
-                GUI::Desktop::the().set_wallpaper(value, false);
-        }
-    } wallpaper_listener;
```

Comment:
What is the rationale behind removing the config listener?

Reply:
None, not sure why I did that! Unfortunately, discovered #11595 whilst putting it back...
Another problem with the way I've designed this is that this would happen:
- Someone sets the wallpaper and provides a path
- Wallpaper is set
- Config string is set
  - Listener picks up config change, and sets the wallpaper (again).

Maybe the desktop _shouldn't_ be responsible for the config path! I'll have to put a bit more thought into this one, certainly don't want to patch an issue by just _removing_ the broken feature :^)

Reply:
May it be possible to temporarily remove the listener, or detect in the listener that this is a recursive call and nothing should be done? E.g. a temporary boolean.

Reply:
Yeah, that's what I was thinking, not sure if there would be a cleaner solution... Something like `bool m_is_currently_changing_wallpaper`.